src/postgres/postgresConnectionStrings.ts

```diff
@@ -37,19 +47,7 @@ export class ParsedPostgresConnectionString extends ParsedConnectionString {
     }
 
     public getEncodedConnectionString(databaseName?: string): string {
-        let connectionString: string = `postgres://`;
-        if (this.username && this.password) {
-            const encodedUsername = encodeURIComponent(this.username);
-            const encodedPassword = encodeURIComponent(this.password);
-            connectionString += `${encodedUsername}:${encodedPassword}@`;
-
-        }
-        connectionString += `${this.hostName}:${this.port}`;
-        if (databaseName) {
-            const encodeDatabaseName = encodeURIComponent(databaseName);
-            connectionString += `/${encodeDatabaseName}`;
-        }
-        return connectionString;
+        return createPostgresConnectionString(this.hostName, this.port, this.username, this.password, this.databaseName ? this.databaseName : databaseName);
```

Comment:
`getEncodedConnectionString` is only called once (in copyConnectionString). I suggest you get rid of it and just call `createPostgresConnectionString` directly

It's a bit confusing how you're doing `this.databaseName ? this.databaseName : databaseName` and calling `createPostgresConnectionString` directly would help alleviate that